shared/src/commonMain/kotlin/io.newm.shared/internal/repositories/WalletRepository.kt

```diff
@@ -17,39 +18,41 @@ internal class WalletRepository(
         cacheService.getWalletConnections()
 
     suspend fun syncWalletConnectionsFromNetworkToDB(): List<WalletConnection> {
-        return try {
+        return runCatching {
             val connections = networkService.getWalletConnections()
             cacheService.deleteAllWalletConnections()
             cacheService.cacheWalletConnections(connections)
-            connections
-        } catch (e: Exception) {
+            cacheService.getWalletConnections().first()
+        }.getOrElse { e ->
             Logger.e(e) { "Error fetching wallet connections from network ${e.cause}" }
-            return emptyList()
+            cacheService.getWalletConnections().first()
+        }.ifEmpty {
+            emptyList()
         }
```

Comment:
nit: instead of ` ...}.ifEmpty { emptyList() }` you can just do `...}.orEmpty()`